### ISSUE TITLE
[Bug]: Focus doesn't land on the expected element in scenario with focusZone

### ISSUE THREAD

**Issue by antrianis** [microsoft/fluentui#25443]
### Library

React / v8 (@fluentui/react)

### System Info

```shell
System:
    OS: Windows 10 10.0.22621
    CPU: (16) x64 11th Gen Intel(R) Core(TM) i9-11900K @ 3.50GHz
    Memory: 37.46 GB / 63.62 GB
  Browsers:
    Edge: Spartan (44.22621.674.0), Chromium (107.0.1418.26), ChromiumDev (108.0.1462.2)
    Internet Explorer: 11.0.22621.1
```

### Reproduction

https://codepen.io/andr33/pen/NWzGWzJ?editors=1111

## Actual Behavior
The focus goes between the two input text fields ignoring the rest of the focusable elements. As soon as you focus manually inside of one of the focus zones, then the focus starts including the rest of the focusable elements.

## Expected Behavior
The focus cycle should include all the items from the beginning.
This seems to be caused by: https://github.com/microsoft/fluentui/pull/24175 

### Requested priority

High

### Are you willing to submit a PR to fix?

no

### Validations

- [X] Check that there isn't already an issue that reports the same bug to avoid creating a duplicate.
- [X] The provided reproduction is a minimal reproducible example of the bug.

**Comment by micahgodbolt** [comment]
The bug is something to do with the combination of List and FocusZone. Render the divs without using list, and it works as expected.

**Comment by antrianis** [comment]
Thanks for looking into it, as discussed, we will need to wait for a fix, since we rely on List specific features.

**Comment by spmonahan** [comment]
This should be fixed when #25331 is merged.
